### PR TITLE
Update dictionary order section

### PR DESCRIPTION
**Pull request by JoeBlackSci**

### CODE REVIEW

File: module01_introduction_to_python/01_06_dictionaries.ipynb [hunk 1]
```diff
@@ -364,38 +364,80 @@
...
+   "source": [
+    "my_dict == my_rev_dict"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "If the order of dictionary elements are important, you can use a specialised form of dictionary called an odered dictionary. Python provides several advanced containers including an OrderedDict as part of the standard library in the [collections](https://docs.python.org/3/library/collections.html) module. "
    ]
```
[the remark this answers is not in the document]
Comment (jemrobinson) [reply module01_introduction_to_python/01_06_dictionaries.ipynb:441]: It might also be worth giving an example of this and splitting this cell into two.

**Review by jemrobinson — CHANGES_REQUESTED** [review]
Thanks for your PR @JoeBlackSci ! I have a couple of comments/suggestions - let me know what you think.

File: module01_introduction_to_python/01_06_dictionaries.ipynb [hunk 2]
```diff
@@ -364,38 +364,80 @@
...
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "\n",
-    "Another consequence of the way dictionaries work is that there's no guaranteed order among the\n",
-    "elements:\n",
-    "\n",
-    "\n"
+    "Python dictionaries will retain the order of the elements as they are defined."
```
[the remark this answers is not in the document]
Comment (JoeBlackSci) [reply module01_introduction_to_python/01_06_dictionaries.ipynb]: This is more concise, I will make the changes.

**Review by JoeBlackSci — COMMENTED** [review]
Hi, I've made the requested changes and I've added an example of ordered dictionaries, just showing how order impacts equality.

**Comment by jemrobinson** [comment]
Hi @JoeBlackSci . This looks great! I possibly didn't quite explain what I meant with the last example - I think it would be helpful to show the reason *why* an OrderedDict might be helpful. For example something like:

```python
my_dict = {"a": "A", "b": "B", "c": "C"}
print(my_dict)
print(my_dict.values())
```

followed by

```python
my_dict["b"] = "second"
print(my_dict)
print(my_dict.values())
```

should change the order to `["a", "c", "b"]` whereas for an `OrderedDict` there will be no change. I'm happy to add this example.

This is probably more useful than the `rev_ordered_dict` example. What do you think?

**Comment by JoeBlackSci** [comment]
Ah, I see what you mean now, however I've just checked on python 3.8, and it appears that altering the value of an element does not cause the order of a Dict to change:

```python
my_dict = {"A": "a", "B": "b", "C": "c", "D": "d"}
print(my_dict)
print(my_dict.values())

my_dict["B"] = "second"
print(my_dict)
print(my_dict.values())
```

output:
```
{'A': 'a', 'B': 'b', 'C': 'c', 'D': 'd'}
dict_values(['a', 'b', 'c', 'd'])
{'A': 'a', 'B': 'second', 'C': 'c', 'D': 'd'}
dict_values(['a', 'second', 'c', 'd'])
```

It appears that the order is maintained even after altering the value.

**Comment by kunjshukla** [comment]
I would like to take this issue up if it is not taken

**Comment by jemrobinson** [comment]
Thank you @kunjshukla but I think @JoeBlackSci is working on this at the moment.

@JoeBlackSci : Given the fact that even a standard dict is maintaining key order, maybe it's best to just drop the discussion of `OrderedDict` entirely?

**Comment by JoeBlackSci** [comment]
Agreed, it's probably out of scope for what most research software will require, and users are likely to encounter the collections module should they ever need it. I'll make a commit dropping the section.

**Pull request merged**